pyomo/core/plugins/transform/zero_sum_propagator.py

```diff
@@ -0,0 +1,81 @@
+"""Transformation to propagate a zero value to terms of a sum."""
+from pyomo.core.base.constraint import Constraint
+from pyomo.core.kernel.numvalue import value
+from pyomo.core.plugins.transform.hierarchy import IsomorphicTransformation
+from pyomo.repn.canonical_repn import generate_canonical_repn
+from pyomo.util.plugin import alias
+
+from six.moves import range
+
+__author__ = "Qi Chen <qichen at andrew.cmu.edu>"
+
+
+class ZeroSumPropagator(IsomorphicTransformation):
```

Comment:
Agreed. Unless I can find a MSc student or an undergrad to do it, I am only likely to hit the cases that are immediately obvious with my models.